src/elements/content-preview/ContentPreview.js

```diff
@@ -354,6 +372,31 @@ class ContentPreview extends React.PureComponent<Props, State> {
         } else if (hasTokenChanged) {
             this.updatePreviewToken();
         }
+
+        if (!prevFile && !!file && !selectedVersion && activeAnnotationId) {
+            // Replace any deep linked annotation fileVersionId with the file's current version because the
+            // current functionality is to load the current version of a deep linked annotation on a previous version
+            // on a hard load.
+            // NOTE: When ContentPreview can support mounting with an older file version this behavior can
+            // likely go away
+            history.replace(`/activity/annotations/${currentFileVersionId}/${activeAnnotationId}`);
```

Comment:
It feels like this logic should live inside a Route/Redirect component, similar to the one at the bottom of SidebarPanels. Maybe it could live in the ActivitySidebar, since that's what it's affecting?